game/src/pregame.rs

```diff
@@ -497,15 +511,16 @@ impl Screensaver {
         }
     }
 
-    fn update(&mut self, rng: &mut XorShiftRng, ctx: &mut EventCtx, map: &Map) {
+    fn update(&mut self, rng: &mut XorShiftRng, ctx: &mut EventCtx, app: &mut App) {
         if ctx.input.nonblocking_is_update_event().is_some() {
             ctx.input.use_update_event();
             let dist_along = Duration::realtime_elapsed(self.started) * SPEED;
             if dist_along < self.line.length() {
+                app.primary.sim.tiny_step(&app.primary.map, &mut app.primary.sim_cb);
```

Comment:
You could move this below the current conditional and remove the `tiny_step` call in `start_bounce`; bit simpler